### PR TITLE
fix(odata-service-writer): Exclude writing ui5 local yaml if service type is CDS in odata service writer

### PR DESCRIPTION
**Pull request by kjose90**
https://github.com/SAP/open-ux-tools/issues/2031

- In `odata-service-writer`, added logic to write the `ui5-local.yaml` file only if the service type is not `cds`, aligning with the existing logic.
- Added test to `odata-service-writer` to check if `ui5-local.yaml` is not written when service type is `cds`

### CODE REVIEW

File: packages/odata-service-writer/src/index.ts [hunk 1]
```diff
@@ -148,7 +148,8 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
         updatePackageJson(paths.packageJson, fs, !!service.metadata);
     }
 
-    if (ui5LocalConfigPath && ui5LocalConfig) {
+    if (!serviceIsCds(service) && ui5LocalConfigPath && ui5LocalConfig) {
```
Comment (IainSAP) [inline packages/odata-service-writer/src/index.ts:151]:
```suggestion
    if (service?.serviceType === Servicetype.CDS && ui5LocalConfigPath && ui5LocalConfig) {
```

No need for the indirection?

File: packages/odata-service-writer/src/data/defaults.ts [hunk 2]
```diff
@@ -64,8 +57,15 @@ export function enhanceData(service: OdataService): void {
...
     setDefaultServiceName(service);
     setDefaultServiceModel(service);
-    setDefaultAnnotationsName(service);
-
+    /**
+     * In the manifest EJS template, annotation names are used to add annotations to the manifest.json.
+     * For CAP projects, annotations are added to the annotations.cds file instead of the manifest.json.
+     * If the service type is EDMX, this function sets the default annotation names to be included in the manifest.json.
+     */
+    if ((service.type ??  ServiceType.EDMX) === ServiceType.EDMX) {
```
Comment (IainSAP) [inline packages/odata-service-writer/src/data/defaults.ts:65]:
```suggestion
    if ((service.type  === ServiceType.EDMX) {
```

File: packages/odata-service-writer/src/index.ts [hunk 3]
```diff
@@ -71,12 +71,13 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     const paths = await findProjectFiles(basePath, fs);
     ensureExists(basePath, ['webapp/manifest.json'], fs);
     enhanceData(service);
-
+    // set isServiceTypeEdmx true if service is EDMX
+    const isServiceTypeEdmx = (service.type ?? ServiceType.EDMX) === ServiceType.EDMX;
```
Comment (IainSAP) [inline packages/odata-service-writer/src/index.ts:75]:
```suggestion
    const isServiceTypeEdmx = service.type === ServiceType.EDMX
```

**Review by IainSAP — COMMENTED** [review]
Some minor comments

File: packages/odata-service-writer/src/index.ts (hunk 3 above)
Comment (kjose90) [reply packages/odata-service-writer/src/index.ts]: Thanks Iain - I moved to default setting to defaults.

**Review by devinea — APPROVED** [review]
changes are clear and clean. 
changeset ✅

**Pull request merged**